rusk-abi/src/lib.rs

```diff
@@ -22,6 +22,12 @@ use dusk_abi::{ContractId, Module};
 use dusk_bls12_381::BlsScalar;
 use dusk_jubjub::{JubJubAffine, JubJubScalar};
 
+/// Transcript label initializations used for proof verifications of
+/// RuskModule
+// This is set outside RuskModule to avoid the need to declare the generic type
+// of store to create a proof
+pub const TRANSCRIPT_LABEL: &'static [u8] = b"dusk-network";
```

Comment:
I don't want that `rusk-abi` become a dependency of circuits or other things like that, so this shouldn't be exposed at all currently.

If the transcript label needs to be taken from somewhere, is not from here.
We might want to create a new issue / PR where we update `rusk-profile` to store and retrieve the transcript label as well.

At the moment I would propose the following solution:
1. Since it's used only on the `verify_proof`, just revert the change. If you want, you can have as a local constant, not exposed, in the `host` file.

2. Make it as "settings" for the host environment, as it is now the Public Parameters. Basically:
```rust
pub struct RuskModule<S> {
    store: S,

    #[cfg(not(target_arch = "wasm32"))]
    pp: &'static dusk_plonk::prelude::PublicParameters,
    #[cfg(not(target_arch = "wasm32"))]
    transcript: &'static [u8],
}
```

Or something like that, since we want to have the exact same flexibility we have with the public parameters.

Reply:
I really don't like this label definition repetition around the code base. This can lead to all sort of silly errors when someone simply uses a different transcript initialization

Reply:
We should simply make that impossible by hardcoding this constant in our circuit definitions/interactions